lib/linked_list.rb

```diff
@@ -7,39 +7,83 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity O(1)
+  # Space complexity O(1)
   def add_first(data)
+    new_node = Node.new(data)
+    if @head = nil
```

Comment:
You use a double = sign to compare value

```suggestion
    if @head == nil
```

Or better yet

```suggestion
    if @head.nil?
```